### ISSUE TITLE
Oscillator problem NOT.

### ISSUE THREAD

**Issue by alto777** [VCVRack/Rack#931]
It's amazing how writing something up to get help from the heavies can solve a problem. I'll spare you the details, but my great discovery in doing was that my 

`json_t *toJson() override {`

gets called periodically. Like at the same times my mystery was manifesting. I was operating under the mistaken assumption that the function was only called when I quit. Or more really, I never even thought about it.

So, live and learn. You'd have enjoyed the longer story, I am sure. Back to the swamp.

**Comment by AndrewBelt** [comment]
Rack autosaves every 15 seconds, but it shouldn't be called much more than that.

**Comment by alto777** [comment]
Yes, I saw that in the log when I was investigating. In fact my problem was happening at the same times, hmm!

Having an inexplicable malfunction drags one into all corners looking for the cause. I found and fixed many errors and potential errors and learned other new things about Rack. And myself. So a blessing in disgrace, I suppose.

One thing I saw and "fixed" raises a question, though. I had a Schmitt trigger and my code sequence implicitly relied on it not firing (again) while within the step. Just to be sure, I buttressed the logic, but:

Is it possible  that my step attention would be interrupted or otherwise running in parallel with the process(es) that handle the button down -> Schmitt trigger process flag? The window for opportunity would be extremely small, but you know what they say about windows...
